### ISSUE TITLE
Calls to clauses should pass state/emit and errors

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/ergo#142]
The current implementation for clause calls do not handle state/emit and errors properly.

**Comment by jeromesimeon** [comment]
Here is an example:
```
bash-3.2$ cat examples/volumediscount/logic7.ergo
namespace org.accordproject.volumediscount

import org.accordproject.contract.*
import org.accordproject.late.*

contract VolumeDiscount over TemplateModel {
// Clause for volume discount
   clause test(rate Double) : Double { return rate + 1.0 }

   clause volumediscount(request VolumeDiscountRequest) : VolumeDiscountResponse {
    if request.netAnnualChargeVolume < contract.firstVolume
		then return new VolumeDiscountResponse{ discountRate: test(contract.firstRate) }
    else if request.netAnnualChargeVolume < contract.secondVolume
		then return new VolumeDiscountResponse{ discountRate: test(contract.secondRate) }
		else return new VolumeDiscountResponse{ discountRate: test(contract.thirdRate) }
  }
}
bash-3.2$ ergo execute --ergo ./examples/volumediscount/logic7.ergo --contractname VolumeDiscount --clausename volumediscount --contract ./examples/volumediscount/contract.json --request ./examples/volumediscount/request.json --state ./examples/volumediscount/state.json --cto ./examples/volumediscount/model.cto 
08:31:40 - info: Logging initialized. 2018-05-11T12:31:40.947Z
08:31:41 - info: {"emit":[],"state":{"$class":"org.accordproject.contract.State","status":"EXECUTORY"},"response":{"discountRate":{"left":{"emit":[],"state":{"$class":"org.accordproject.contract.State","status":"EXECUTORY"},"response":3.8}},"$class":"org.accordproject.volumediscount.VolumeDiscountResponse"}}
```

**Comment by jeromesimeon** [comment]
Addressed in #323 as part of work for #277
